lib/Renard/Curie/Component/PageDrawingArea.pm

```diff
@@ -276,13 +277,49 @@ fun on_key_press_event_cb($window, $event, $self) {
 	}
 }
 
+=method setup_scroll_bindings
+
+	method setup_scroll_bindings()
+
+Sets up the signals to capture scroll events on this component.
+
+=cut
+method setup_scroll_bindings () {
+	$self->scrolled_window->signal_connect ( scroll_event => \&on_scroll_event_cb, $self );
+}
+
+=func on_scroll_event_cb
+
+	func on_scroll_event_cb($window, $event, $self)
+
+Callback that responds to specific scroll events and dispatches the associated
+handlers.
+
+=cut
+fun on_scroll_event_cb($window, $event, $self) {
+	if ( $event->state == 'control-mask' && $event->direction eq 'smooth') {
+		my ($delta_y, $delta_x) =  $event->get_scroll_deltas();
```

Comment:
`$delta_y` and `$delta_x` are swapped. See [gdk_event_get_scroll_deltas ()](https://developer.gnome.org/gdk3/stable/gdk3-Events.html#gdk-event-get-scroll-deltas):

```c
gboolean
gdk_event_get_scroll_deltas (const GdkEvent *event,
                             gdouble *delta_x,
                             gdouble *delta_y);
```